### PR TITLE
use `if` instead of short-circuiting `&&` or `||`

### PR DESCRIPTION
**Pull request by nsajko**
Should result in finer-grained code coverage.

### CODE REVIEW

File: src/FixedSizeArrays.jl [hunk 1]
```diff
@@ -85,8 +87,9 @@ axes_are_one_based(axes) = all(isone ∘ first, axes)
 
 function FixedSizeArray{T,N}(src::AbstractArray{S,N}) where {T,N,S}
     axs = axes(src)
-    axes_are_one_based(axs) ||
+    if !axes_are_one_based(axs)
         throw(DimensionMismatch("source array has a non-one-based indexing axis"))
```
Comment (giordano) [inline src/FixedSizeArrays.jl:91]: This line isn't exercised at the moment, but can probably be done in a separate PR

**Review by giordano — APPROVED** [review]

File: src/FixedSizeArrays.jl (hunk 1 above)
Comment (nsajko) [reply src/FixedSizeArrays.jl:91]: xref #32

**Review by giordano — APPROVED** [review]
Finally all green! 🥳
